server/datastore/mysql/software.go

```diff
@@ -3172,15 +3172,25 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 	bySoftwareID := make(map[uint]*hostSoftware)
 
 	hostSoftwareInstalls, err := hostSoftwareInstalls(ds, ctx, host.ID)
+	installQuarantineSet := make(map[uint]*hostSoftware)
 	if err != nil {
 		return nil, nil, err
 	}
 	for _, s := range hostSoftwareInstalls {
 		if _, ok := bySoftwareTitleID[s.ID]; !ok {
-			bySoftwareTitleID[s.ID] = s
+			if opts.OnlyAvailableForInstall || opts.IncludeAvailableForInstall {
+				bySoftwareTitleID[s.ID] = s
+			} else {
+				installQuarantineSet[s.ID] = s
+			}
 		} else {
-			bySoftwareTitleID[s.ID].LastInstallInstalledAt = s.LastInstallInstalledAt
-			bySoftwareTitleID[s.ID].LastInstallInstallUUID = s.LastInstallInstallUUID
+			if opts.OnlyAvailableForInstall || opts.IncludeAvailableForInstall {
+				bySoftwareTitleID[s.ID].LastInstallInstalledAt = s.LastInstallInstalledAt
+				bySoftwareTitleID[s.ID].LastInstallInstallUUID = s.LastInstallInstallUUID
+			} else {
+				installQuarantineSet[s.ID].LastInstallInstalledAt = s.LastInstallInstalledAt
+				installQuarantineSet[s.ID].LastInstallInstallUUID = s.LastInstallInstallUUID
+			}
```

Comment:
Sorry this changed, this is debatable here. Do we want to pull up the software installer and potentially associate it with the installed software? The method that pulls up what osquery returns doesn't find matching software installers. It relies on this set